zhmcclient/_manager.py

```diff
@@ -537,57 +547,74 @@ def invalidate_cache(self):
 
     def _try_optimized_lookup(self, filter_args):
         """
-        Try to optimize the lookup by checking whether the filter arguments
-        specify the property that is used as the last segment in the resource
-        URI, with a plain string as match value (i.e. not using regular
-        expression matching).
+        Try to find a resource in an optimized way when the filter arguments
+        allow for that.
 
-        If so, the lookup is performed by constructing the resource URI from
-        the specified filter argument, and by issuing a Get Properties
-        operation on that URI, returning a single resource object.
+        The following cases of optimized filtering are supported:
 
-        Parameters:
+        - A single filter on the resource name, that does not use regular
+          expression matching.
+        - A single filter on the resource object/element ID, that does not use
+          regular expression matching.
 
+        Returns `None` if the filter arguments do not meet these optimization
+        criteria, or if they do but no resource was found.
+
+        Parameters:
           filter_args (dict):
             Filter arguments. For details, see :ref:`Filtering`.
 
         Returns:
-
-          resource object, or `None` if the optimization was not possible.
+          resource object, or `None` if the the filter arguments did not meet
+            the optimization criteria, or if they did but no resource was found.
         """
-        if filter_args is None or len(filter_args) != 1 or \
-                self._oid_prop not in filter_args:
+        if filter_args is None or len(filter_args) != 1:
             return None
 
-        oid_match = filter_args[self._oid_prop]
-        if not isinstance(oid_match, six.string_types) or \
-                not re.match(r'^[a-zA-Z0-9_\-]+$', oid_match):
-            return None
+        if self._name_prop in filter_args:
 
-        # The match string is a plain string (not a reg.expression)
+            name_match = filter_args[self._name_prop]
+            if not isinstance(name_match, six.string_types) or \
```

Comment:
the condition

 not isinstance(name_match, six.string_types) or REGEXP_SPECIAL_CHAR.search(name_match):

could go into a helper method

Reply:
And how would that help readability?

Reply:
I tried to find a suitable name for the helper method and failed.
The check for not being a string and the check for being a regexp string have no common semantics.
So this would be a helper function with no well defined semantics.